contracts/Dispatcher.sol

```diff
@@ -276,12 +276,16 @@ contract Dispatcher is IbcDispatcher, Ownable {
 
         // For XXXX => vIBC direction, SC needs to verify the proof of membership of TRY_PENDING
         // For vIBC initiated channel, SC doesn't need to verify any proof, and these should be all empty
-        if (
-            counterparty.channelId != 0 ||
-            bytes(counterparty.version).length != 0 ||
-            proof.proofHeight != 0 ||
-            proof.proof.length != 0
-        ) {
+        bool isChanOpenTry = false;
+        if (counterparty.channelId == bytes32(0) && bytes(counterparty.version).length == 0) {
+            // ChanOpenInit with unknow conterparty
+        } else if (counterparty.channelId != bytes32(0) && bytes(counterparty.version).length != 0) {
+            // this is the ChanOpenTry; counterparty must not be zero-value
+            isChanOpenTry = true;
+        } else {
+            revert('Invalid counterparty');
```

Comment:
Good to see `revert` is used here, so I can learn what that is, and what's the difference between `require` and `revert`
When search online (correct me if I read them wrong).

- `require`: take caller's gas, meaning caller makes a mistake therefore they experience gas loss.
- `revert`: do not take caller's gas, meaning this is a situation typically Solidity contract has buggy logic.   Since this is SC's bug, do not take caller's gas.

==================================
If above understanding is correct (please consult more experienced SOL developer in our team), 
- I think `revert` shouldn't be used here, since this is a caller's error.  (We should take their gas).
- We probably wish to re-examine our SOLs for `require` and `revert` usage, to make sure they are correctly used.

Reply:
Let's do that in another PR? I added an issue #13. Custom errors seem better than error strings

Reply:
If I understand @qinlz2's concern we could simply replace 'revert' with 'require' to prevent spamming this call with invalid counterparty data?

Reply:
nm, this explains it pretty well: https://medium.com/blockchannel/the-use-of-revert-assert-and-require-in-solidity-and-the-new-revert-opcode-in-the-evm-1a3a7990e06e

revert == require but for more complex conditionals like we have here. It allows refunding unspent gas which is good.

assert is for hard invariants - situations which should never occur